core/src/main/java/org/mule/runtime/core/internal/processor/simple/RemoveFlowVariableProcessor.java

```diff
@@ -22,7 +25,13 @@
 
   @Override
   protected PrivilegedEvent removeProperty(PrivilegedEvent event, String propertyName) {
-    return PrivilegedEvent.builder(event).removeVariable(propertyName).build();
+    if (event.getVariables().containsKey(propertyName)) {
+      return PrivilegedEvent.builder(event).removeVariable(propertyName).build();
+    } else {
+      LOGGER.warn("There is no variable named '%s'. You might want to check the expression in remove-variable component in: %s",
```

Comment:
`You might want to check the expression ` -> `Check the 'variableName' parameter`